packages/cli/src/bin/plugins/app-proxy.ts

```diff
@@ -0,0 +1,142 @@
+ *
+ * This plugin sets up proxy rules for API and bundle requests to the Fusion app backend.
+ * It also serves the app configuration and manifest based on the app key and version.
+ *
+ * @param options - Configuration options for the plugin.
+ * @param options.proxyPath - The base path for the proxy.
+ * @param options.target - The target URL for the proxy.
+ * @param options.app - Information about the app including key and version.
+ * @param options.app.key - The key identifying the app.
+ * @param options.app.version - The version of the app.
+ * @param options.generateConfig - A function that generates the app configuration.
+ * @param options.generateConfig.return - A promise that resolves to the app configuration.
+ * @param options.generateManifest - A function that generates the app manifest.
+ * @param options.generateManifest.return - A promise that resolves to the app manifest.
+ * @returns A Vite plugin object.
+ *
+ * @remarks
+ * This plugin sets up proxy rules for API and bundle requests to the Fusion app backend.
+ * It also serves the app configuration and manifest based on the app key and version.
+ *
+ * @example
+ * ```typescript
+ * const plugin = appProxyPlugin({
+ *   proxyPath: '/app-proxy',
+ *   target: 'https://fusion-s-apps-ci.azurewebsites.net/',
+ *   app: { key: 'my-app', version: '1.0.0' },
+ *   generateConfig: async () => ({}), // app config object
+ *   generateManifest: async () => ({}), // app manifest object
+ * });
+ *
+ * // api calls
+ * fetch('/app-proxy/apps/my-app/builds/1.0.0/config'); // wil generate app config by provided function
+ * fetch('/app-proxy/apps/my-app/builds/0.0.9/config'); // wil proxy to the target, since version does not match
+ * fetch('/app-proxy/apps/other-app/builds/1.0.0/config'); // wil proxy to the target, since app key does not match
+ *
+ * // asset calls
+ * fetch('/app-proxy/bundles/my-app/builds/1.0.0/index.js'); // wil generate bundle by provided function
+ * fetch('/app-proxy/bundles/my-app/builds/0.0.9/index.js'); // wil proxy to the target, since version does not match
+ * ```
+ */
+export const appProxyPlugin = (options: {
+    proxyPath: string;
+    target: string;
+    app: { key: string; version: string };
+    generateConfig: () => Promise<AppConfig>;
+    generateManifest: () => Promise<ApplicationManifest>;
+}): Plugin => {
+    const { app, proxyPath, generateConfig, generateManifest } = options;
+    return {
+        name: 'fusion:app-proxy',
+        apply: 'serve',
+        config(config) {
+            config.server ??= {};
+            config.server.proxy = {
+                // proxy all api calls to the fusion apps backend
+                [`${proxyPath}/apps`]: {
+                    target: 'https://fusion-s-apps-ci.azurewebsites.net/',
+                    changeOrigin: true,
+                    secure: false,
+                    rewrite: (path) => path.replace(/^\/app-proxy/, ''),
```

Comment:
Should the replace here use the proxyPath? sins this is an input to the plugin?

Reply:
yupp, had it in the back of my mind :) but to trigger happy